scenario_player/utils/configuration/token.py

```diff
@@ -0,0 +1,74 @@
+import datetime
+import pathlib
+import uuid
+from typing import Optional, Union
+
+import structlog
+
+from scenario_player.exceptions.config import TokenConfigurationError
+from scenario_player.utils.configuration.base import ConfigMapping
+
+log = structlog.get_logger(__name__)
+
+
+class TokenConfig(ConfigMapping):
+    def __init__(self, loaded_yaml: dict, data_path: pathlib.Path):
+        super(TokenConfig, self).__init__(loaded_yaml.get("token", {}))
+        self._token_id = uuid.uuid4()
+        self._name = None
+        self._token_file = data_path.joinpath("token.info")
+        self.validate()
+
+    @staticmethod
+    def assert_option(expression, err: Optional[Union[str, Exception]] = None):
+        """Assert the given expression and raise a NodeConfigurationError if it fails."""
+        if isinstance(err, str):
+            err = TokenConfigurationError(err)
+        return ConfigMapping.assert_option(expression, err)
+
+    def validate(self):
+        """Validate the configuration section.
+
+        Asserts that:
+
+            * Mutually exclusive options are not truthy
+        """
+        mutual_exclusive_ops = ("address", "reuse")
+        if all(option in self.dict for option in mutual_exclusive_ops):
+            self.assert_option(
+                bool(self["address"]) != self["reuse"],
+                f"Token settings {mutual_exclusive_ops} are mutually exclusive.",
+            )
+
+    @property
+    def name(self):
+        if not self._name:
+            now = datetime.datetime.now()
+            self._name = self.get(
+                "name", f"Scenario Test Token {self._token_id!s} {now:%Y-%m-%dT%H:%M}"
+            )
+        return self._name
+
+    @property
+    def address(self):
+        return self.get("address")
+
+    @property
+    def block(self):
```

Comment:
What does `block` do?

Reply:
I dont get the question? :thinking:

Reply:
No I mean I don't know this setting. I don't think it exists ;)

Reply:
Aah..gotcha. Fixed!